ops/framework.py

```diff
@@ -1061,13 +1061,17 @@ def __init__(self, parent: Object, attr_name: str):
 
     if TYPE_CHECKING:
         @typing.overload
-        def __getattr__(self, key: Literal['on']) -> ObjectEvents:  # type: ignore
+        def __getattr__(self, key: Literal['on']) -> ObjectEvents:
+            pass
+
+        @typing.overload
+        def __getattr__(self, key: str) -> Any:
```

Comment:
Do you know why this overload is needed when it's the same types as the actual `def __getattr__` below? Is the main signature not used as a fallback when there are overloads?

Reply:
Got it, thanks for the explanation.